### PR TITLE
Avoid policy conflicts

### PR DESCRIPTION
**Pull request by Snawoot**
Postfix allows use of multiple maps for `smtp_tls_policy_maps` configuration directive, so policies lookup is performed across multiple sources. Current code generates config with `may` policy set explicitly for testing domains in list, because this list alone can't conclude any better policy. Such behavior is undesired because of these reasons:

- Other maps may set more restrictive policies for those domains. For example, [policies based on MTA-STS lookups](https://github.com/Snawoot/postfix-mta-sts-resolver#postfix-mta-sts-resolver). In such cases value of first matched map will be result of lookup, depending on order they are declared.
- Postfix configuration may have default policy other than `may` level.

There are [good example](https://tools.ietf.org/html/rfc8461#section-2) of similar conflict resolution in MTA-STS RFC 8461:

>    The primary motivation of MTA-STS is to provide a mechanism for
>    domains to ensure transport security even when deploying DNSSEC is
>    undesirable or impractical.  However, MTA-STS is designed not to
>    interfere with DANE deployments when the two overlap; in particular,
>    senders who implement MTA-STS validation MUST NOT allow MTA-STS
>    Policy validation to override a failing DANE validation.
> 

In this case, if list contains no enforcing policy for domain, then config generator shouldn't override policy for this domain.

Actual PR replaces `may` policy of testing domains with comment line containing domain name and reason why it is not included in generated map. Resulting line looks like this:

```
#gmail.com                         undefined due to testing policy
```

Such behavior was chosen for two reasons:

- Make absence of testing domains more transparent to user.
- Preserve current flow of config generation in code.

Related tests are already adjusted according to this change.

Signed-off-by: Vladislav Yarmak <vladislav@vm-0.com>

### CODE REVIEW

File: starttls-policy/starttls_policy/configure.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def _policy_for_domain(domain, tls_policy, max_domain_len):
         line += " secure match="
         line += ":".join(tls_policy.mxs)
     elif tls_policy.mode == "testing":
-        line += "may "
+        line = '#' + line + "undefined due to testing policy"
```
Comment (sydneyli) [inline starttls-policy/starttls_policy/configure.py]: very small nit: dbl quote to match context and space after `#`

Comment (Snawoot) [reply starttls-policy/starttls_policy/configure.py]: Fixed, thanks!

Comment (Snawoot) [reply starttls-policy/starttls_policy/configure.py]: Also space added

**Comment by Snawoot** [comment]
@sydneyli ok, finally sorted things out. I think now its ready.

**Comment by Snawoot** [comment]
Hi! Any updates?

**Pull request merged**